src/systems/filecoin_markets/retrieval_market/retrieval_flow_v0.mmd

```diff
@@ -0,0 +1,67 @@
+sequenceDiagram
+
+  participant Client as RetrievalClient
+  participant Provider as RetrievalProvider
+  participant Node
+  participant BSC as BlockStore (Client)
+  participant BSP as BlockStore (Provider)
+  participant StorageMiner as Storage Miner
+
+  opt Discovery
+    Client ->> Node : Query Chain For Providers w/ PieceID
+    activate Node
+    Node -->> Client : Providers With PieceID
+    deactivate Node
+  end
+
+  Client ->> Provider : Send Query for Piece
+  activate Provider
+  Provider -->> Client : Send information about Piece (price, size, etc)
+  deactivate Provider
+  Client ->> Node : Create/Activate Payment Channel
+  activate Node
+  Node -->> Client : Message For Channel Gets On Chain
+  deactivate Node
+  Client ->> Provider : Send RetrievalDealProposal
+  activate Provider
+  Provider -->> Client : Send RetrievalDealResponse, accepting
+  activate Client
+  opt Unsealing
+  Provider ->> StorageMiner : Unseal Sector
+  activate StorageMiner
+  StorageMiner -->> Provider : Unsealed Piece Data
+  deactivate StorageMiner
+  Provider ->> BSP: Write Piece Blocks to BlockStore
+  activate BSP
+  BSP -->> Provider : Blocks written
+  deactivate BSP
+  end
+  loop Retreive In Pieces
+    Provider ->> Client : Request Payment
+    activate Client
+    Client ->> Node : Create Voucher On Payment Channel
+    activate Node
+    Node -->> Client : Message For Voucher Gets On Chain
+    deactivate Node
+    Client -->> Provider : Send Payment
```

Comment:
wouldn't the client send payment on chain in this case, and the provider redeem the voucher on payment channel (ie get paid) from the chain accordingly?

Reply:
sorry bad wording -- this happens through the payment channel -- should read something like
`Client ->> Provider : Send Voucher For Provider to Redeem`

Reply:
awesome. That makes sense to me